### DISCUSSION TITLE
brew update/upgrade warning and error

### DISCUSSION THREAD

**Discussion by robrobin** [Homebrew/discussions#3859] [Getting started] [answered]
when i run "brew update"，i got response:

> brew update
Warning: git 2.38.1 is already installed and up-to-date.
To reinstall 2.38.1, run:
  brew reinstall git
/usr/local/Homebrew/Library/Homebrew/cmd/update.sh: line 37: : command not found
Warning: No remote 'origin' in /usr/local/Homebrew, skipping update!
/usr/local/Homebrew/Library/Homebrew/cmd/update.sh: line 37: : command not found
Warning: No remote 'origin' in /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask, skipping update!
/usr/local/Homebrew/Library/Homebrew/cmd/update.sh: line 37: : command not found
Warning: No remote 'origin' in /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core, skipping update!
Error: update-report should not be called directly!

when i run "brew upgrade", i got response:

> brew upgrade
Running `brew update --auto-update`...
Warning: git 2.38.1 is already installed and up-to-date.
To reinstall 2.38.1, run:
  brew reinstall git
/usr/local/Homebrew/Library/Homebrew/cmd/update.sh: line 37: : command not found
Warning: No remote 'origin' in /usr/local/Homebrew, skipping update!
/usr/local/Homebrew/Library/Homebrew/cmd/update.sh: line 37: : command not found
Warning: No remote 'origin' in /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask, skipping update!
/usr/local/Homebrew/Library/Homebrew/cmd/update.sh: line 37: : command not found
Warning: No remote 'origin' in /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core, skipping update!
Error: update-report should not be called directly!

brew config
> HOMEBREW_VERSION: 3.6.8-10-g73d535a
ORIGIN: https://github.com/Homebrew/brew
HEAD: 73d535a206d86c06e6ebc266351a53e28692dace
Last commit: 6 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 0e1d40f22ad76fdeb4d96bddc5da6f4eb60b8f1a
Core tap last commit: 2 days ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 14.0.0 build 1400
Git: 2.38.1 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6-x86_64
CLT: 14.1.0.0.1.1666437224
Xcode: N/A

brew doctor

> Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!
Warning: No Cask quarantine support available: unknown reason.

i had reinstall brew many times, and the problem is not solved.

**Comment by SMillerDev** [comment, marked as answer]
Your system somehow does not have a git binary anymore. Try running `xcode-select --install`.

**Reply by robrobin** [reply]
Thanks!
Git was disturbed by some encryption software. When i removed the encryption software, it get right!
